src/repl.ts

```diff
@@ -239,9 +255,27 @@ function _eval(service: Service, state: EvalState, input: string) {
     state.output = output;
   }
 
-  return changes.reduce((result, change) => {
-    return change.added ? exec(change.value, state.path) : result;
+  const result: any = changes.reduce((result, change) => {
+    if (change.added) {
+      if (
+        /\bawait\b/.test(change.value) &&
+        service.options.experimentalReplAwait
+      ) {
+        const result = processTopLevelAwait(change.value);
+        if (result !== null) {
+          awaitPromise = true;
+          return exec(result, state.path);
+        }
+      }
+      return exec(change.value, state.path);
+    }
+    return result;
   }, undefined);
```

Comment:
I was thinking about why `.reduce` is being used here.  Hypothetically, if there is more than one item in the `changes` array to be executed, then could there be multiple promises which must be awaited?

With the synchronous implementation, all but the last value can be discarded.  But with async, we might need to await a promise from the first change before executing the second.

Rather than proving that multiple promises are impossible, it seems easier to merely assume it might be possible and await them.  This means refactoring from `.reduce` into a `for` loop.

Reply:
Fixed in last implementation.

Reply:
_eval implementation could have remained using .reduce, as now the execution iteration is moved to `evalCodeInternal` consumers. However I still prefer to use `for of` since it was less code overall.